javascripts/discourse/components/featured-homepage-topics.js

```diff
@@ -1,83 +1,116 @@
 import discourseComputed from "discourse-common/utils/decorators";
 import Component from "@ember/component";
+import I18n from "I18n";
 import { inject as service } from "@ember/service";
 import Topic from "discourse/models/topic";
 import { defaultHomepage } from "discourse/lib/utilities";
 
+const FEATURED_CLASS = "featured-homepage-topics";
+
 export default Component.extend({
   classNameBindings: ["featured-homepage-topics"],
   router: service(),
   titleElement: null,
   featuredTagTopics: null,
 
+  _getBanner() {
+    if (!(this.get("isDestroyed") || this.get("isDestroying"))) {
+      let sortOrder = settings.sort_by_created ? "created" : "activity";
+      if (settings.featured_tag) {
+        this.store
+          .findFiltered("topicList", {
+            filter: "latest",
+            params: {
+              tags: [`${settings.featured_tag}`],
+              order: sortOrder,
+            },
+          })
+          .then((topicList) => {
+            let featuredTagTopics = [];
+
+            topicList.topics.forEach((topic) =>
+              featuredTagTopics.push(Topic.create(topic))
+            );
+
+            this.set(
+              "featuredTagTopics",
+              featuredTagTopics.slice(0, settings.number_of_topics)
+            );
+          });
+      }
+    }
+  },
+
+  _checkClass() {
+    if (!this.showHere) {
+      document.querySelector("body").classList.remove(FEATURED_CLASS);
+    }
+  },
+
   init() {
     this._super(...arguments);
-    let sortOrder = settings.sort_by_created ? "created" : "activity";
-    if (settings.featured_tag) {
-      this.store
-        .findFiltered("topicList", {
-          filter: "latest",
-          params: {
-            tags: [`${settings.featured_tag}`],
-            order: sortOrder,
-          },
-        })
-        .then((topicList) => {
-          let featuredTagTopics = [];
+    this._getBanner();
+    this._checkClass();
+  },
 
-          topicList.topics.forEach((topic) =>
-            topic.image_url ? featuredTagTopics.push(Topic.create(topic)) : ""
-          );
+  didInsertElement() {
+    this.appEvents.on("page:changed", this, "_checkClass");
```

Comment:
When you register an event listener in didInsertElement, you should then remove it in willDestroyElement

Reply:
though, does it need to listen to page:changed at all? 🤔